src/runtime/objmodel.cpp

```diff
@@ -961,6 +964,7 @@ Box* typeLookup(BoxedClass* cls, llvm::StringRef attr, GetattrRewriteArgs* rewri
                 assert(rewrite_args->obj == obj_saved);
             } else {
                 rewrite_args->obj = rewrite_args->rewriter->loadConst((intptr_t)base, Location::any());
+                rewrite_args->obj_cls_guarded = true;
```

Comment:
Is this true?  I think you might be allowed to to change the metaclass of a base class...  Maybe this could be `obj_cls_guarded = base->cls->is_constant`?